lib/spack/spack/stage.py

```diff
@@ -490,7 +490,9 @@ def print_errors(errors):
             print_errors(errors)
 
             self.fetcher = self.default_fetcher
-            raise fs.FetchError(err_msg or 'All fetchers failed', None)
+            if not err_msg:
+                err_msg = 'All fetchers failed for {0}'.format(self.name)
+            raise fs.FetchError(err_msg, None)
```

Comment:
```suggestion
            default_msg =  'All fetchers failed for {0}'.format(self.name)
            raise fs.FetchError(err_msg or default_msg, None)
```

Reply:
Done, even though I don't find this more legible. T0o much implicit function of a boolean operator